bwc-test/src/test/java/org/opensearch/security/bwc/SecurityBackwardsCompatibilityIT.java

```diff
@@ -0,0 +1,344 @@
+    }
+
+    public void testBasicBackwardsCompatibility() throws Exception {
+        String round = System.getProperty("tests.rest.bwcsuite_round");
+
+        if (round.equals("first") || round.equals("old")) {
+            assertPluginUpgrade("_nodes/" + CLUSTER_NAME + "-0/plugins");
+        } else if (round.equals("second")) {
+            assertPluginUpgrade("_nodes/" + CLUSTER_NAME + "-1/plugins");
+        } else if (round.equals("third")) {
+            assertPluginUpgrade("_nodes/" + CLUSTER_NAME + "-2/plugins");
+        }
+    }
+
+    /**
+     * Tests backward compatibility by created a test user and role with DLS, FLS and masked field settings. Ingests
+     * data into a test index and runs a matchAll query against the same.
+     */
+    public void testDataIngestionAndSearchBackwardsCompatibility() throws Exception {
+        String round = System.getProperty("tests.rest.bwcsuite_round");
+        String index = "test_index";
+        if (round.equals("old")) {
+            createTestRoleIfNotExists(TEST_ROLE);
+            createUserIfNotExists(TEST_USER, TEST_PASSWORD, TEST_ROLE);
+            createIndexIfNotExists(index);
+        }
+        ingestData(index);
+        searchMatchAll(index);
+    }
+
+    @SuppressWarnings("unchecked")
+    private void assertPluginUpgrade(String uri) throws Exception {
+        Map<String, Map<String, Object>> responseMap = (Map<String, Map<String, Object>>) getAsMap(uri).get("nodes");
+        for (Map<String, Object> response : responseMap.values()) {
+            List<Map<String, Object>> plugins = (List<Map<String, Object>>) response.get("plugins");
+            Set<String> pluginNames = plugins.stream().map(map -> (String) map.get("name")).collect(Collectors.toSet());
+
+            final Version minNodeVersion = minimumNodeVersion();
+
+            assertThat(pluginNames, hasItem("opensearch_security"));
+        }
+    }
+
+    /**
+     * Ingests data into the test index
+     * @param index index to ingest data into
+     */
+
+    private void ingestData(String index) throws IOException {
+        StringBuilder bulkRequestBody = new StringBuilder();
+        ObjectMapper objectMapper = new ObjectMapper();
+        for (Song song : Song.SONGS) {
+            Map<String, Map<String, String>> indexRequest = new HashMap<>();
+            indexRequest.put("index", new HashMap<>() {
+                {
+                    put("_index", index);
+                }
+            });
+            bulkRequestBody.append(objectMapper.writeValueAsString(indexRequest) + "\n");
+            bulkRequestBody.append(objectMapper.writeValueAsString(song.asJson()) + "\n");
+        }
+
+        Response response = RestHelper.makeRequest(
+            testUserAuthClient,
+            "POST",
+            "_bulk",
+            RestHelper.toHttpEntity(bulkRequestBody.toString())
+            // TODO: Resolve the following line
+            // List.of(encodeBasicHeader(TEST_USER, TEST_PASSWORD))
```

Comment:
Found this line, was this working as expected or a debug statement to remove?

Reply:
This was for debugging. Will remove.

Reply:
I see you've already removed it as part of refactoring. Resolving this.